### PR TITLE
React & Redux group project - Space Travelers' Hub

### PR DESCRIPTION
**Pull request by Atifzada**
### In this milestone I:

- Configured & set up react and redux.
- Fetched data and updated the Redux store.
- Rendered the data in the UI.
- Wrote actions and reducers for booking rockets and joining missions.
- Wrote actions and reducers for canceling rockets and leaving missions.
- Used conditional components rendering.
- Created My Profile section.
- Tested all components.

Note: Due to political unrest in my country (pakistan) there is internet outage issue so It was suggested by student success team to do the project solely. So I did it all by myself. Attaching the screenshots of the conversation with students success team as in evidence.

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  moduleNameMapper: {
+    '^@/(.*)$': '<rootDir>/src/$1',
+    '\\.(css|less|sass|scss)$': 'identity-obj-proxy',
+    '\\.(jpg|jpeg|png|gif|svg)$': '<rootDir>/src/tests/fileMock.js', // Add this line to handle image imports
+  },
+  testMatch: ['**/tests/**/*.test.(js|jsx)'],
+  testPathIgnorePatterns: ['/node_modules/', '/public/'],
+  collectCoverageFrom: ['src/**/*.{js,jsx}'],
+  testEnvironment: 'jsdom',
+};
```
Comment (AmaduKamara) [inline jest.config.js:15]: - [ ] This file is causing the test from running successfully. Kindly fix the error it's giving so the tests can run successfully 👍🏽

Screenshot 2023-05-19 at 8 28 36 AM

**Review by AmaduKamara — CHANGES_REQUESTED** [review]
# Status: Required Changes ♻️ ♻️ ♻️

Hi @Atifzada,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✅ You used Gitflow correctly 👍🏽
✅ The Linter checks passed successfully 👍🏽
✅ Used Redux and react correctly 👍🏽
✅ The app runs fine 👍🏽
✅ Navigation works fine 👍🏽
✅ Rockets data rendered correctly 👍🏽

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by AmaduKamara — CHANGES_REQUESTED** [review]
Kindly work on the changes and the test to run successfully 👍🏽

File: jest.config.js (hunk 1 above)
Comment (Atifzada) [reply jest.config.js:15]: Sorry, I forget to delete this file. Thank you for the helpful review

File: src/styles/Navbar.module.css [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  font-weight: 600;
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  gap: 1rem;
+  padding-right: 2rem;
+}
+
+.navlinks a {
+  color: #037bfe;
```
Comment (danmainah) [inline src/styles/Navbar.module.css:31]: - [ ] @Atifzada great work implementing the changes requested, however, you only missed one requirement, the color of the link as previously highlighted by the previous reviewer, check the image below, the link's color is supposed to match the design given.
image

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @Atifzada , 

Great work on making the changes requested by a previous reviewer 👏  👏 👏
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
✔️ Linter setup correctly and all checks are passing
✔️ Good design on the rockets page.


 # Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

- [ ] [optional] @Atifzada great work on your tests they are all working, I would recommend you add tests for pure functions in redux directory.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/styles/Navbar.module.css (hunk 2 above)
Comment (Atifzada) [reply src/styles/Navbar.module.css:31]: @danmainah Thank you for the review but I already add the color which was asked for. Anyhow I will again look into it.

Comment (danmainah) [reply src/styles/Navbar.module.css]: @Atifzada the nav items didn't have te change of color still the same, hope you have implemented the change

**Review by ShadyShawkat — APPROVED** [review]
## Hi @Atifzada, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight
- All linters are passing ✔️ 
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 
- Your readme looks professional ✔️ 

Image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**